lib/jsonrpc.ml

```diff
@@ -554,27 +554,41 @@ let get name dict =
     raise (Malformed_method_request str)
   | Some v -> v
 
-let call_of_string str =
+let version_and_call_of_string str =
   match of_string str with
   | Dict d ->
-    let name = match get "method" d with String s -> s | _ -> raise (Malformed_method_request str) in
-    let params =
+    let name =
+      match get "method" d with
+      | String s -> s
+      | _ -> raise (Malformed_method_request "method")
```

Comment:
I like specified what failed, but I think having also the rest of the string in here would make it easier to debug.
What about adding:
```ocaml
let malformed_method_error loc = Malformed_method request (Printf.sprintf "\"%s\" in %s" loc str) in 
```
and then `raise (malformed_method_error "method")` here and similarly in the calls below